core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -462,22 +461,22 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
               .get_info<sycl::info::kernel_device_specific::max_sub_group_size>(
                   q.get_device(),
                   sycl::range<3>(m_team_size, m_vector_size, 1));
-      if (max_sg_size % m_vector_size != 0) {
+      auto final_vector_size = std::min<int>(m_vector_size, max_sg_size);
+      if (max_sg_size % final_vector_size != 0) {
         std::stringstream out;
         out << "The maximum subgroup size (" << max_sg_size
             << ") for this kernel is not divisible by the vector_size ("
-            << m_vector_size << "). Choose a smaller vector_size!\n";
+            << final_vector_size << "). Choose a smaller vector_size!\n";
```

Comment:
Can you elaborate on why this needs to be revisited? Should we explain that the vector size might have been capped?

Reply:
The error message will only be triggered if m_vector_size was smaller than max_sg_size in the first place (since you use the min). So one doesn't need to necessarily choose a smaller vector size, just something which is divisible (which could be larger). That said, how would this be ever triggered? Is max_sg_size not a power of two? We kinda have that requirement for vector length in team policy.